planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/start_planner_module.cpp

```diff
@@ -335,6 +336,57 @@ bool StartPlannerModule::hasCollisionWithDynamicObjects() const
   return !isSafePath();
 }
 
+bool StartPlannerModule::isInsideLanelets() const
+{
+  const auto & current_pose = planner_data_->self_odometry->pose.pose;
+  const auto vehicle_footprint = autoware_utils::transform_vector(
+    vehicle_info_.createFootprint(), autoware_utils::pose2transform(current_pose));
+
+  lanelet::BasicPolygon2d footprint_polygon;
+  for (const auto & point : vehicle_footprint) {
+    footprint_polygon.push_back({point.x(), point.y()});
+  }
+
+  // Find lanelets that intersect with the vehicle footprint
+  const auto & lanelets_distance_pair = lanelet::geometry::findWithin2d(
+    planner_data_->route_handler->getLaneletMapPtr()->laneletLayer, footprint_polygon, 0.0);
```

Comment:
Do we need to check all the lanes, not just the lane departure check lane?

Reply:
Ah I see lane_departure_checker_ also check all lanelets
https://github.com/tier4/autoware_universe/blob/c509904eb2b0fb38f4645860508dd79073331af9/planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/shift_pull_out.cpp#L108-L110